src/mono/mono/mini/mini-amd64.c

```diff
@@ -623,6 +669,12 @@ add_valuetype (MonoMethodSignature *sig, ArgInfo *ainfo, MonoType *type,
 #ifdef TARGET_WIN32
 	add_valuetype_win64 (sig, ainfo, type, is_return, gr, fr, stack_size);
 #else
+#ifdef MONO_ARCH_HAVE_SWIFTCALL
+	if (is_return && sig->pinvoke && mono_method_signature_has_ext_callconv (sig, MONO_EXT_CALLCONV_SWIFTCALL)) {
+		add_return_valuetype_swiftcall (sig, ainfo, type, gr, fr, stack_size);
```

Comment:
Is it possible to move this to `get_call_info`? This should be invoked only for wrappers and could be simplified without changes in `add_valuetype`.

Reply:
It's possible, changed!